src/KM_MapEditor.pas

```diff
@@ -355,10 +396,6 @@ procedure TKMMapEditor.MouseUp(Button: TMouseButton; aOverMap: Boolean);
                                 //Actual change was made in UpdateStateIdle, we just register it is done here
                                 fTerrainPainter.MakeCheckpoint;
                               end;
-                cmObjects:    begin
-                                gTerrain.Land[P.Y,P.X].Obj := 255; //Delete object
-                                fTerrainPainter.MakeCheckpoint;
-                              end;

```

Comment:
It looks like this will never be invoked, since in TKMapEdInterface.MouseDown we invoke RightClick_Cancel if Right mouse btn was clicked, where we reset cursor mode